I1463 open shift connector - ready for the review (#1080)
codenvy/che-docs: src/main/pages/che-7/extensions/proc_connecting-source-code-from-guthub-to-a-openshift-component-using-openshift-connector.adoc

---
title: Connecting source code from GitHub to a OpenShift Component
keywords:
tags: []
sidebar: che_7_docs
permalink: che-7/connecting-source-code-from-guthub-to-a-openshift-component-using-openshift-connector/
folder: che-7/extensions
summary:
---
// using-openshift-connector-in-eclipse-che

[id="connecting-source-code-from-guthub-to-a-openshift-component-using-openshift-connector_{context}"]

> 

Reviewer: ```suggestion
```

= Connecting source code from GitHub to an OpenShift Component using OpenShift Connector

When the user has a Git-stored source code that is wanted for further development, it is more efficient to deploy it directly from the Git repository into the Openshift Connector Component.

This chapter describes how to obtain the content from the Git repository and connect it with a {prod-short}-developed OpenShift Component.

.Prerequisites
* Have a running  {prod-short} workspace.
* Be logged in to the OpenShift cluster using the OpenShift Connector.

.Procedure

> To make changes to your GitHub component, you need to clone the repository into {prod-short} to obtain this source code. To do so:

Reviewer: ```suggestion
To make changes to your GitHub component, clone the repository into {prod-short} to obtain this source code:
```

> . In the {prod-short} main screen, open the *Command Palette* by pressing F1.

Reviewer: ```suggestion
. In the {prod-short} main screen, open the *Command Palette* by pressing kbd:[F1].
```

> . Type the `Git Clone` command in the *Command Palette* and hit btn:[Enter].

Reviewer: ```suggestion
. Type the `Git Clone` command in the *Command Palette* and press kbd:[Enter].
```

. Provide the GitHub URL and select the destination for the deployment.
. Add source-code files to your Project by clicking the btn:[Add to workspace] button.

For additional information about cloning Git repository, see:

ifeval::["{project-context}" == "che"]
link:https://www.eclipse.org/che/docs/che-7/version-control/#accessing-a-git-repository-via-https_version-control[Accessing a Git repository via HTTPS]
endif::[]

ifeval::["{project-context}" == "crw"]
link:https://access.redhat.com/documentation/en-us/red_hat_codeready_workspaces/2.0/html-single/end-user_guide/index#accessing-a-git-repository-via-https_version-control[Accessing a Git repository via HTTPS]

> endif::[]

Reviewer: No downstream stuff in upstream! Use attributes for URLs if needed.

Author: I will try to answer this in Bj session.
